src/java.base/share/classes/java/io/CharArrayWriter.java

```diff
@@ -97,10 +98,8 @@ public void write(int c) {
      *          of the given array
      */
     public void write(char[] c, int off, int len) {
-        if ((off < 0) || (off > c.length) || (len < 0) ||
-            ((off + len) > c.length) || ((off + len) < 0)) {
-            throw new IndexOutOfBoundsException();
-        } else if (len == 0) {
+        Objects.checkFromToIndex(off, len, c.length);
```

Comment:
shouldn't that be Objects.checkFromToIndex(off, off+len, c.length);

Reply:
It looks like all the occurrences of `checkFromToIndex()` should instead be `checkFromIndexSize()`.

Reply:
Indeed there are at least four regression test failures with this as-is:

- java/io/InputStream/ReadParams.java
- java/io/OutputStream/WriteParams.java
- java/io/Writer/WriteParams.java
- java/nio/channels/FileChannel/Transfer2GPlus.java

Reply:
> Indeed there are at least four regression test failures with this as-is:
> 
> * java/io/InputStream/ReadParams.java
> * java/io/OutputStream/WriteParams.java
> * java/io/Writer/WriteParams.java
> * java/nio/channels/FileChannel/Transfer2GPlus.java

The java/io tests run in tier2 and I think the GHA only run tier1.

Reply:
Fixed along with copyright year

Reply:
The regression tests now pass.